components/TextFields/src/MDCTextInputControllerLegacyDefault.m

```diff
@@ -0,0 +1,1261 @@
+
+@property(nonatomic, copy) NSString *errorAccessibilityValue;
+@property(nonatomic, copy, readwrite) NSString *errorText;
+@property(nonatomic, copy) NSString *previousLeadingText;
+
+@property(nonatomic, strong) UIColor *previousPlaceholderColor;
+
+@property(nonatomic, strong) UIFont *customLeadingFont;
+@property(nonatomic, strong) UIFont *customPlaceholderFont;
+@property(nonatomic, strong) UIFont *customTrailingFont;
+
+@end
+
+@implementation MDCTextInputControllerLegacyDefault
+
+@synthesize characterCounter = _characterCounter;
+@synthesize characterCountMax = _characterCountMax;
+@synthesize characterCountViewMode = _characterCountViewMode;
+@synthesize errorColor = _errorColor;
+@synthesize floatingPlaceholderScale = _floatingPlaceholderScale;
+@synthesize textInput = _textInput;
+@synthesize underlineViewMode = _underlineViewMode;
+
+// TODO: (larche): Support in-line auto complete.
+
+- (instancetype)init {
+  self = [super init];
+  if (self) {
+    [self commonMDCTextInputControllerDefaultInitialization];
+  }
+
+  return self;
+}
+
+- (instancetype)initWithCoder:(NSCoder *)aDecoder {
+  self = [super init];
+  if (self) {
+    [self commonMDCTextInputControllerDefaultInitialization];
+
+    _activeColor =
+        [aDecoder decodeObjectForKey:MDCTextInputControllerDefaultActiveColorKey];
+    _characterCounter =
+        [aDecoder decodeObjectForKey:MDCTextInputControllerDefaultCharacterCounterKey];
+    _characterCountMax =
+        [aDecoder decodeIntegerForKey:MDCTextInputControllerDefaultCharacterCountMaxKey];
+    _characterCountViewMode =
+        [aDecoder decodeIntegerForKey:MDCTextInputControllerDefaultCharacterCountViewModeKey];
+    _disabledColor = [aDecoder decodeObjectForKey:MDCTextInputControllerDisabledColorKey];
+    _errorColor = [aDecoder decodeObjectForKey:MDCTextInputControllerDefaultErrorColorKey];
+    _floatingEnabled = [aDecoder decodeBoolForKey:MDCTextInputControllerDefaultFloatingEnabledKey];
+    _floatingPlaceholderColor =
+        [aDecoder decodeObjectForKey:MDCTextInputControllerDefaultFloatingPlaceholderColorKey];
+    _floatingPlaceholderScale =
+        [aDecoder decodeObjectForKey:MDCTextInputControllerDefaultFloatingPlaceholderScaleKey];
+    _inlinePlaceholderColor =
+        [aDecoder decodeObjectForKey:MDCTextInputControllerDefaultInlinePlaceholderColorKey];
+    _textInput = [aDecoder decodeObjectForKey:MDCTextInputControllerDefaultTextInputKey];
+    _normalColor =
+        [aDecoder decodeObjectForKey:MDCTextInputControllerDefaultNormalColorKey];
+    _underlineViewMode = (UITextFieldViewMode)
+        [aDecoder decodeIntegerForKey:MDCTextInputControllerDefaultUnderlineViewModeKey];
+  }
+  return self;
+}
+
+- (instancetype)initWithTextInput:(UIView<MDCTextInput> *)textInput {
+  self = [self init];
+  if (self) {
+    _textInput = textInput;
+  }
+
+  // This should happen last because it relies on the state of a ton of properties.
+  [self setupInput];
```

Comment:
Consider moving inside if (self) block?
Usually I don't expect to see anything expect return self; after the if block.